src/plugins/discover/public/application/components/sidebar/discover_sidebar.tsx

```diff
@@ -112,6 +124,34 @@ export function DiscoverSidebar({
     ]
   );
 
+  const paginate = useCallback(() => {
+    setPageSize(
+      Math.max(PAGINATION_SIZE, Math.min(pageSize + PAGINATION_SIZE * 0.5, unpopularFields.length))
+    );
+  }, [setPageSize, pageSize, unpopularFields]);
+
+  useEffect(() => {
+    if (scrollContainer && unpopularFields) {
+      const { clientHeight, scrollHeight } = scrollContainer;
+
+      if (clientHeight === scrollHeight && pageSize < unpopularFields.length) {
+        // in case of there is so much height available that there's no scrolling at initial rendering
+        // pagination starts after the first bulk of rendering
+        setTimeout(() => paginate(), 100);
```

Comment:
@flash1293 was't completely done here, so I've adapted the code that caused your troubles. But I'm not certain that this could not cause other issues. 

I tried to fix the following case: there are 50 fields of 3000 rendered, but due to zooming or a very large vertical monitor, there is no scrolling necessary for those 50 rendered fields. Then you this kind of pagination wouldn't work, you would stick to 50 fields. any ideas, or just set to 100?